db/hooks.php

```diff
@@ -0,0 +1,45 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Definition of MFA sub-plugins (factors).
+ *
+ * @package     tool_mfa
+ * @author      Liam Kearney <liam@sproutlabs.com.au>
+ * @copyright   Catalyst IT
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die();
+
+// Hooks are Totara specific.
+if (class_exists('\totara_core\hook\base')) {
+    $watchers = [
+        [
+            'hookname' => '\core\hook\after_require_login',
```

Comment:
```suggestion
            'hookname' => core\hook\after_require_login::class,
```
Would this also work? The benefit here is some IDEs such as VSCode and PHPStorm pick up on these symbols, and if they resolve to a real class, it will display the relevant tooltip / jump-to / definition, which could be an extra check to ensure a real class is being referenced. Otherwise, if it exists as a string, it might not be as useful. 

That said, this is Totara specific and I am not well versed in these new hooks declarations, and the way you have it might be more consistent with how they've set up their examples.

It could similarly apply for after_config and the `totara_core\hook\base::class` check.

What are your thoughts here?

Reply:
Good suggestion, absolutely all for magic classnames!

I've converted the callable syntax to use `array<class-string, string>` instead as well.